akka-stream/src/main/scala/akka/stream/impl/Sinks.scala

```diff
@@ -441,3 +445,102 @@ private[akka] final class ReducerState[T, R](val collector: java.util.stream.Col
   def finish(): R = collector.finisher().apply(reduced)
 }
 
+/**
+ * INTERNAL API
+ */
+final private[stream] class LazySink[T, M](sinkFactory: T ⇒ Future[Sink[T, M]], zeroMat: () ⇒ M) extends GraphStageWithMaterializedValue[SinkShape[T], Future[M]] {
```

Comment:
All additions to the AST complicate matters internally quite a lot, adding new edge cases everywhere. Things should not be added there, instead the existing tools should be used to model behavior whenever possible.

Btw, that does not exclude us from adding `Source.whatever(...).lazy` to the DSL, since that will be simply:

```scala
def lazy: Source[T] = Source.fromGraph(new LazySource(this))
```

or something like that.